R/parallel-chat.R

```diff
@@ -54,7 +66,13 @@
 #' type_person <- type_object(name = type_string(), age = type_number())
 #' parallel_chat_structured(chat, prompts, type_person)
 #' \dontshow{ellmer:::vcr_example_end()}
-parallel_chat <- function(chat, prompts, max_active = 10, rpm = 500) {
+parallel_chat <- function(
+  chat,
+  prompts,
+  max_active = 10,
+  rpm = 500,
+  on_error = "return"
```

Comment:
Would this be better matching the structure of `httr2::req_perform_parallel()`, as in:

```
#' @param on_error Character string specifying how to handle errors:
#'   `"continue"` (default) continues processing all prompts, returning NA for failed ones (maintains input length),
#'   `"return"` stops at first error and returns only successful results processed before the error, printing the error message,
#'   `"stop"` stops on first error like the original function.
```

(from https://github.com/tidyverse/ellmer/pull/705/files#diff-3c0ac75881c575e3ff5a68fc8b9b54c8c012f4dbd5b6986d02f9f12daae2dc69R193)